File: data/src/main/java/org/oppia/data/backends/gae/model/Outcome.java
package org.oppia.data.backends.gae.model;

import com.squareup.moshi.Json;

import java.util.List;

public class Outcome {

  /*
   * Parent Hierarchy: ExplorationContainer -> Exploration -> State -> InteractionInstance -> Outcome
   * Parent Hierarchy: ExplorationContainer -> Exploration -> State -> InteractionInstance -> AnswerGroup -> Outcome
   */

  //TODO: ImmutableList<> and List<> error
  //ImmutableList<> does not work but List<> works

  @Json(name = "dest")
  public String dest;

  @Json(name = "refresher_exploration_id")
  public String refresherExplorationId;

  @Json(name = "feedback")
  public SubtitledHtml feedback;

  @Json(name = "param_changes")
  public List<ParamChange> paramChanges;

  @Json(name = "missing_prerequisite_skill_id")
  public String missingPrerequisiteSkillId;

  @Json(name = "labelled_as_correct")
  public boolean labelledAsCorrect;

}

File: data/src/main/java/org/oppia/data/backends/gae/model/Outcome.kt
package org.oppia.data.backends.gae.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Outcome(

  @Json(name = "dest") val dest: String?,
  @Json(name = "refresher_exploration_id") val refresherExplorationId: String?,
  @Json(name = "feedback") val feedback: SubtitledHtml?,
  @Json(name = "param_changes") val paramChanges: List<ParamChange>?,
  @Json(name = "missing_prerequisite_skill_id") val missingPrerequisiteSkillId: String?,
  @Json(name = "labelled_as_correct") val isLabelledAsCorrect: Boolean?

)
